File: Karume-lab/power_of_two.py
def is_power_of_two(num: int) -> bool:
    """
    Summary:
        Determines whether a given integer is a power of two.
    Parameters:
        num : int
            The number to check if it's a power of two.
    Returns:
        bool
            Returns True if the number is a power of two, False otherwise.
    """
    if num == 0:
        return False
    while num != 1:
        if num % 2 != 0:
            return False
        num //= 2

    return True

File: mamesareagan/power_of_two.c
#include <stdio.h>
#include <stdbool.h>

// Function to check if a number is a power of 2
bool isPowerOfTwo(unsigned int n) {
        return (n > 0) && ((n & (n - 1)) == 0);
}

int main() {
	unsigned int num;

	num = 8;
	
    	printf("%u => %s\n", num, isPowerOfTwo(num) ? "true" : "false");

	num = 6;
	printf("%u => %s\n", num, isPowerOfTwo(num) ? "true" : "false");
	return 0;
}

